pkg/operator/ceph/object/spec.go

```diff
@@ -158,11 +158,25 @@ func (c *clusterConfig) makeRGWPodSpec(rgwConfig *rgwConfig) (v1.PodTemplateSpec
 		ServiceAccountName: serviceAccountName,
 	}
 
+	opsLogFile := ""
+	if opsLogSidecar := c.store.Spec.Gateway.OpsLogSidecar; opsLogSidecar != nil {
+		// Generate ops log file name based on rgw daemon ID
+		opsLogFile := fmt.Sprintf("ops-log-%s.log", getDaemonName(rgwConfig))
+
+		// Add the side-car container named ops-log
+		podSpec.Containers = append(podSpec.Containers,
+			*controller.RgwOpsLogSidecarContainer(opsLogFile,
+				c.clusterInfo.Namespace, *c.clusterSpec,
+				opsLogSidecar.Resources))
+	}
+
 	// If the log collector is enabled we add the side-car container
 	if c.clusterSpec.LogCollector.Enabled {
 		shareProcessNamespace := true
 		podSpec.ShareProcessNamespace = &shareProcessNamespace
-		podSpec.Containers = append(podSpec.Containers, *controller.LogCollectorContainer(getDaemonName(rgwConfig), c.clusterInfo.Namespace, *c.clusterSpec))
+		podSpec.Containers = append(podSpec.Containers,
+			*controller.LogCollectorContainer(getDaemonName(rgwConfig),
+				c.clusterInfo.Namespace, *c.clusterSpec, opsLogFile))
```

Comment:
Based on my reading, I think Parth's comment here (https://github.com/rook/rook/pull/14646/files#r1887158232) needs addressed. `opsLogFile` seems to be the filename but not the full file path. I think `opsLogFile` should be something like `path.Join(config.VarLogCephDir, opsLogFile)` IIUC